### PR TITLE
fix(ui): remember last spawn tool and worktree

### LINKED ISSUE #486: New session tool and worktree reset to Settings instead of the last pick
### What happened

Creating a session, picking a tool and worktree on the `n` form or the quick bar, then opening `n` or the quick bar again reseeds those two fields from Settings (`default_tool`, `worktree sessions`) and the group's worktree default. The last pick lives only in the form/quick state for that opening.

`openForm` always calls `defaultToolSelection()` and `groupWorktree(...)`. `openQuickMode` does the same for the tab cycle and `alt+w`. Tab / shift+tab change `toolIndex` and `worktree` in memory; nothing writes them back. The next `n` or `space` starts over.

People who spawn several sessions in a row (opencode + worktree, then another) have to retoggle both every time. Settings still own the long-lived defaults; this is about the last spawn in this manager process, so a run of new sessions keeps the tool and worktree you just used.

### What I expected

The next `n` form and the next quick-bar spawn start on the tool and worktree of the last new session in this process. Settings remain the fallback when there is no last pick yet (first spawn after launch, or after the last tool is hidden).

### Investigate

Whether to persist that last pick only in memory for the running manager, or in the store like Settings. In-memory matches "this batch of sessions"; store would survive a quit. Group worktree `on`/`off`/`inherit` still needs a rule so a group's explicit default is not silently overridden. Confirming a spawn is what should cache the pick; cancelling the form should not.

### Steps

1. Press `n`, set tool to opencode, worktree on, create the session.
2. Press `n` again.
3. Tool and worktree are the Settings defaults, not opencode / on.

Same with the quick bar: on a group row, `tab` / `shift+tab`, send, open the bar again.

### agent-manager version

0.36.0

### Operating system

macOS

### tmux version

tmux 3.7b

### Which agent CLI

Grok Build

### Agent CLI version

grok 1.0.25 (f7e67d6988e2) [stable]

### PR DESCRIPTION
**Pull request by YoanWai**
## What changed

The `n` form and the quick bar now open on the tool and worktree of the last confirmed new session in this process.

## Why

Closes #486

Spawning several sessions in a row meant retoggling tool and worktree every time. The last pick lived only for that opening; the next `n` or space started from Settings again.

### Required behavior

After a spawn succeeds, the next `n` form and the next quick-bar spawn start on that tool and worktree. Settings remain the fallback when this process has no last pick yet, or the last tool has been hidden. A group's explicit worktree on/off still seeds that group, including children that inherit it. Cancelling the form or a spawn that fails leaves the last pick unchanged. The Settings screen still shows the stored defaults.

### Why this approach

The pick lives in two fields on the running Model, which matches a batch of sessions in one manager process. An empty tool means there has been no successful spawn yet; one resolver handles group inheritance and the remembered worktree choice. Settings stay the long-lived default. The write happens only after `spawnSession` succeeds, including a launch-hint retry, so a cancelled or refused form does not stick.

The change is the shared spawn picker. It holds for every enabled CLI and every platform the manager builds.

## How it was verified

- `env -u TMUX TMUX_TMPDIR=/tmp/amr507 go test -race ./...` passes.
- `gofmt -l .` prints nothing; `go vet ./...` and `go build ./...` pass.
- Both form and quick-bar tests launch a fixture CLI, fail a second launch on a missing executable, then install a fake CLI through the launch dialog and verify that only the successful retry updates the next picker.
- Cross-compilation passes with CGO disabled for darwin/amd64, darwin/arm64, linux/amd64, and linux/arm64.
- The shared picker has no provider-specific branches. Tests run locally on macOS with fixture CLIs and real tmux/git; live provider sessions and runtime execution on the other release targets were not exercised.

## Visual evidence

**Not applicable because:** the change is the seed of the next form after a spawn. The tests open `n` and the quick bar twice and assert the last pick.

## Summary by CodeRabbit

- **New Features**
  - New sessions and quick-launch actions remember tool and worktree choices from the last successful launch.
  - Remembered selections carry between form-based and quick launches.
  - Group-specific worktree settings take precedence over remembered choices.

- **Bug Fixes**
  - Cancelled or failed launches no longer change remembered selections.
  - Successful install retries now save the selected choices.
  - If a remembered tool is unavailable, the configured settings default is used.
  - Settings continue to display the configured default independently of recent launch choices.

### CODE REVIEW

**Comment by YoanWai** [comment]
@coderabbitai full review

File: internal/ui/quick_test.go [hunk 1]
```diff
@@ -540,6 +540,114 @@ func TestQuickSpawnUsesTabCycledTool(t *testing.T) {
...
+		t.Fatalf("quick spawn: %q", m.errBar.text)
+	}
+	m.applyCmd(t, cmd)
+	closeQuick(m)
+
+	m.openQuickMode()
+	if !m.quickWorktreeOn() {
+		t.Fatal("next quick bar should seed worktree from the last spawn")
+	}
+}
```
[the remark this answers is not in the document]
Comment (YoanWai) [reply internal/ui/quick_test.go:601]: Reviewed against 0827f33 using the repository's finding gate:

1. Reachable: quickSpawn at internal/ui/quick.go:171 resolves the selected group's pick; quickWorktreeOn at :219 uses the same spawnWorktreeDefault resolver. A prior remembered on plus an explicit group off is reachable.
2. Real: no defect reproduced. The traced path is spawnWorktreeDefault (internal/ui/settings.go:88-93) returning false for explicit off before reading remembered state, followed by quickSpawn:179 recording that false after success. A subsequent inheriting group returns the recorded false at settings.go:96-97. The suggested scenario already behaves as requested. TestFormLastWorktreeYieldsToGroupDefault exercises the shared resolver's explicit/inherited precedence; quick spawn and retry tests cover recording successful picks.
3. In scope: the picker is in scope, but this is a coverage suggestion, not a demonstrated defect in the diff. The cited spawnAndRemember helper no longer exists.
4. Cost: zero production lines are needed. Another multi-launch integration test would duplicate the covered resolver and recording behavior without fixing a defect.

Rejected at gate 2; no code change.

File: internal/ui/settings.go [hunk 2]
```diff
@@ -95,6 +93,9 @@ func (m *Model) groupWorktree(group string) bool {
 			return false
 		}
 	}
+	if m.lastSpawnTool != "" {
+		return m.lastSpawnWorktree
```
[the remark this answers is not in the document]
Comment (YoanWai) [reply internal/ui/settings.go]: Fixed in 1193202.

1. Reachable: `spawnWorktreeDefault` at `internal/ui/settings.go:96` used `lastSpawnWorktree` whenever `lastSpawnTool` was non-empty. `openForm` (`form.go:295`) and `quickWorktreeOn` (`quick.go:219`) both call it, so a hidden last tool still seeded worktree on.
2. Real: `TestFormHiddenLastToolFallsBackToSettings` and `TestQuickHiddenLastToolFallsBackToSettings` failed on that path before the change (`form.worktree` / `quickWorktreeOn()` stayed true after hiding the last CLI with Settings worktree off).
3. In scope: this branch's remembered spawn defaults.
4. Cheap: use the remembered worktree only when that tool is still in `enabledToolNames()`, otherwise `defaultWorktree()`.

**Comment by YoanWai** [comment]
@coderabbitai full review

File: internal/ui/form.go [hunk 3]
```diff
@@ -535,14 +535,20 @@ func (m *Model) submitForm() (tea.Model, tea.Cmd) {
...
 		// it; the dialog releases its images once no install can still
 		// spawn it. An error reported in the bar leaves the form up, and
 		// the prompt still names them.
-		m.reportLaunchError(err, func() error {
-			return m.spawnSession(toolName, name, dir, group, prompt, autoNamed, worktree)
-		})
+		m.reportLaunchError(err, spawn)
 		return m, nil
 	}
 	// New sessions start as starting, which attention excludes; clear so
```
[the remark this answers is not in the document]
Comment (YoanWai) [reply internal/ui/form.go:554]: Rejected at gate 2; no code change.

1. Reachable: `submitForm` at `internal/ui/form.go:537-543` spawns with `formWorktreeOn()` and remembers `m.form.worktree`. `quickSpawn` at `internal/ui/quick.go:170-179` does the same with `quickWorktreeOn()` vs `pickWorktree`. A non-repo dir with the picker still true is reachable (`TestFormWorktreeGatedInNonRepoDir` seeds worktree on, points the form at a plain dir, and still launches).
2. Real: the split is the specified pick, not a defect. Issue #486 and this PR remember the last confirmed picker choice so a run of spawns keeps the tool and worktree the operator set. Capability gating already forces the created session off (`formWorktreeOn` / `quickWorktreeOn` → `spawnSession`). Recording that forced off would wipe a worktree-on pick after any plain-dir spawn, and the next git-dir `n` would start off. That is the reset #486 asked to stop.
3. In scope: this branch owns the remember path.
4. Cost: two call sites. Left unchanged because gate 2 failed.

`TestFormWorktreeGatedInNonRepoDir` already pins that a gated submit creates a plain session. The next capable dir should still see the pick.

**Comment by YoanWai** [comment]
@coderabbitai full review
